diff -Naur a/nvptx-tools/nvptx-as.c b/nvptx-tools/nvptx-as.c
--- a/nvptx-tools/nvptx-as.c	2020-08-24 14:38:02.000000000 +0200
+++ b/nvptx-tools/nvptx-as.c	2020-08-25 10:13:43.328254878 +0200
@@ -51,6 +51,10 @@
 #define X_OK 1
 #endif
 
+#ifndef CUDA_ARCH
+# define CUDA_ARCH "sm_30"
+#endif
+
 #ifndef DIR_SEPARATOR
 #  define DIR_SEPARATOR '/'
 #endif
@@ -1047,7 +1051,7 @@
   FILE *out = stdout;
   bool verbose __attribute__((unused)) = false;
   int verify = -1;
-  const char *smver = "sm_30";
+  const char *smver = CUDA_ARCH;
 
   int o;
   int option_index = 0;
