diff -Naur SuiteSparse_config/SuiteSparse_config.mk SuiteSparse_config.new/SuiteSparse_config.mk
--- a/SuiteSparse_config/SuiteSparse_config.mk	2020-04-08 22:42:59.000000000 +0200
+++ b/SuiteSparse_config/SuiteSparse_config.mk	2020-08-26 14:46:05.071378745 +0200
@@ -241,13 +241,6 @@
                 MAGMA_INC     = -I/opt/magma-2.4.0/include/
                 MAGMA_LIB     = -L/opt/magma-2.4.0/lib/ -lmagma
         NVCC          = $(CUDA_PATH)/bin/nvcc
-        NVCCFLAGS     = -Xcompiler -fPIC -O3 \
-                            -gencode=arch=compute_30,code=sm_30 \
-                            -gencode=arch=compute_35,code=sm_35 \
-                            -gencode=arch=compute_50,code=sm_50 \
-                            -gencode=arch=compute_53,code=sm_53 \
-                            -gencode=arch=compute_53,code=sm_53 \
-                            -gencode=arch=compute_60,code=compute_60
     endif
 
     #---------------------------------------------------------------------------
