diff -Naur a/fem/src/modules/DCRComplexSolve.F90 b/fem/src/modules/DCRComplexSolve.F90
--- a/fem/src/modules/DCRComplexSolve.F90	2020-11-10 20:52:44.000000000 +0100
+++ b/fem/src/modules/DCRComplexSolve.F90	2022-11-24 16:10:14.162346186 +0100
@@ -433,7 +433,8 @@
 !------------------------------------------------------------------------------
       LOGICAL :: FirstTime = .TRUE., stat
       REAL(KIND=dp), POINTER :: Hwrk(:,:,:)
-
+      INTEGER :: i
+      
       SAVE FirstTime, Hwrk
 !------------------------------------------------------------------------------
       IF ( FirstTime ) THEN
@@ -485,7 +486,8 @@
 !------------------------------------------------------------------------------
       LOGICAL :: FirstTime = .TRUE., stat
       REAL(KIND=dp), POINTER :: Hwrk(:,:,:)
-
+      INTEGER :: i
+      
       SAVE FirstTime, Hwrk
 !------------------------------------------------------------------------------
       IF ( FirstTime ) THEN
@@ -1074,6 +1076,7 @@
 !------------------------------------------------------------------------------
       LOGICAL :: FirstTime = .TRUE., stat
       REAL(KIND=dp), POINTER :: Hwrk(:,:,:)
+      INTEGER :: i
 
       SAVE FirstTime, Hwrk
 !------------------------------------------------------------------------------
@@ -1374,6 +1377,7 @@
 !------------------------------------------------------------------------------
       LOGICAL :: FirstTime = .TRUE., stat
       REAL(KIND=dp), POINTER :: Hwrk(:,:,:)
+      INTEGER :: i
 
       SAVE FirstTime, Hwrk
 !------------------------------------------------------------------------------
@@ -1426,6 +1431,7 @@
 !------------------------------------------------------------------------------
       LOGICAL :: FirstTime = .TRUE., stat
       REAL(KIND=dp), POINTER :: Hwrk(:,:,:)
+      INTEGER :: i
 
       SAVE FirstTime, Hwrk
 !------------------------------------------------------------------------------
@@ -1727,6 +1733,7 @@
 !------------------------------------------------------------------------------
       LOGICAL :: FirstTime = .TRUE., stat
       REAL(KIND=dp), POINTER :: Hwrk(:,:,:)
+      INTEGER :: i
 
       SAVE FirstTime, Hwrk
 !------------------------------------------------------------------------------
@@ -1779,6 +1786,7 @@
 !------------------------------------------------------------------------------
       LOGICAL :: FirstTime = .TRUE., stat
       REAL(KIND=dp), POINTER :: Hwrk(:,:,:)
+      INTEGER :: i
 
       SAVE FirstTime, Hwrk
 !------------------------------------------------------------------------------
