diff -Naur a/dolfin/io/HDF5Interface.cpp b/dolfin/io/HDF5Interface.cpp
--- a/dolfin/io/HDF5Interface.cpp	2019-04-26 09:43:04.000000000 +0200
+++ b/dolfin/io/HDF5Interface.cpp	2022-11-24 16:19:12.876488137 +0100
@@ -282,9 +282,13 @@
   }
 
   H5O_info_t object_info;
+#if H5_VERS_MINOR < 12
   H5Oget_info_by_name(hdf5_file_handle, group_name.c_str(), &object_info,
-                      lapl_id);
-
+		      lapl_id);
+#else
+  H5Oget_info_by_name(hdf5_file_handle, group_name.c_str(), &object_info,
+		      H5O_INFO_BASIC, lapl_id);
+#endif
   // Close link access properties
   status = H5Pclose(lapl_id);
   dolfin_assert(status != HDF5_FAIL);
