--- a/src/mpid/ch3/channels/mrail/src/gen2/ibv_cuda_util.c	2020-06-01 23:48:34.000000000 +0200
+++ b/src/mpid/ch3/channels/mrail/src/gen2/ibv_cuda_util.c	2020-09-08 00:49:56.470027061 +0200
@@ -841,7 +841,7 @@
         if (rdma_check_cuda_attribute) {
             cuda_err = cudaPointerGetAttributes (&attributes, buffer);
             if (cuda_err == cudaSuccess) {
-                is_dev = (attributes.memoryType == cudaMemoryTypeDevice) ? 1 : 0;
+                is_dev = (attributes.type == cudaMemoryTypeDevice) ? 1 : 0;
             }
         }
     } else {
--- a/src/mpid/pamid/src/mpidi_util.c	2020-09-08 00:52:20.144015984 +0200
+++ b/src/mpid/pamid/src/mpidi_util.c	2020-09-08 00:52:53.222011464 +0200
@@ -1980,7 +1980,7 @@
 
       if (e != cudaSuccess)
           result = false;
-      else if (cuda_attr.memoryType ==  cudaMemoryTypeDevice)
+      else if (cuda_attr.type ==  cudaMemoryTypeDevice)
           result = true;
       else
           result = false;
